src/blocked.c

```diff
@@ -627,6 +655,8 @@ void handleClientsBlockedOnKeys(void) {
                  * module is trying to accomplish right now. */
                 serveClientsBlockedOnKeyByModule(rl);
             } else {
+                /* Unblock all XREADGROUP clients of this deleted key */
+                unblockDeletedStreamReadgroupClients(rl);
```

Comment:
what if the key was deleted, but then added as a new type before we got here (i.e. `dbOverwrite`), in that case `o!=NULL` and we need to handle it in the above case too.

let's add a test for this case